Newly published docker images for v1.9.1?

Hey friends, thanks for maintaining this awesome gem! ❤️ 

I just noticed that new docker images were published for v1.9.1 19 hours ago (which is the version released back in January):
https://github.com/rake-compiler/rake-compiler-dock/pkgs/container/rake-compiler-dock-image/versions

And as a result (at least I think so) I started to seeing failures in the Herb CI, without updating the gem or the Ruby version:
https://github.com/marcoroth/herb/actions/runs/16625692383/job/47041649259

```
Digest: sha256:1231e49cbe478478749928c69373c34644066d98cdeb21545125561977369ed7
Status: Downloaded newer image for ghcr.io/rake-compiler/rake-compiler-dock-image:1.9.1-mri-x86_64-linux-gnu
rbenv: version `3.4.1' is not installed (set by /home/runner/work/herb/herb/.ruby-version)
rake aborted!
Command failed with status (1): [docker run -v /home/runner/work/herb/herb:/home/runner/work/herb/herb -e UID\=1001 -e GID\=118 -e USER\=runner -e GROUP\=docker -e GEM_PRIVATE_KEY_PASSPHRASE -e SOURCE_DATE_EPOCH -e ftp_proxy -e http_proxy -e https_proxy -e RCD_HOST_RUBY_PLATFORM\=x86_64-linux -e RCD_HOST_RUBY_VERSION\=3.4.1 -e RCD_IMAGE\=ghcr.io/rake-compiler/rake-compiler-dock-image:1.9.1-mri-x86_64-linux-gnu -w /home/runner/work/herb/herb --rm -i ghcr.io/rake-compiler/rake-compiler-dock-image:1.9.1-mri-x86_64-linux-gnu runas sigfw bash -c bundle\ --local\ \&\&\ rake\ native:x86_64-linux-gnu\ gem\ RUBY_CC_VERSION\=\'3.4.1:3.3.7:3.2.6:3.1.6:3.0.7\']
/home/runner/work/herb/herb/vendor/bundle/ruby/3.4.0/gems/rake-compiler-dock-1.9.1/lib/rake_compiler_dock/starter.rb:90:in 'block in RakeCompilerDock::Starter.exec'
/home/runner/work/herb/herb/vendor/bundle/ruby/3.4.0/gems/rake-compiler-dock-1.9.1/lib/rake_compiler_dock/starter.rb:44:in 'Array#each'
/home/runner/work/herb/herb/vendor/bundle/ruby/3.4.0/gems/rake-compiler-dock-1.9.1/lib/rake_compiler_dock/starter.rb:44:in 'RakeCompilerDock::Starter.exec'
/home/runner/work/herb/herb/vendor/bundle/ruby/3.4.0/gems/rake-compiler-dock-1.9.1/lib/rake_compiler_dock/starter.rb:14:in 'RakeCompilerDock::Starter.sh'
/home/runner/work/herb/herb/vendor/bundle/ruby/3.4.0/gems/rake-compiler-dock-1.9.1/lib/rake_compiler_dock.rb:47:in 'RakeCompilerDock.sh'
/home/runner/work/herb/herb/Rakefile:97:in 'block (3 levels) in <top (required)>'
/home/runner/work/herb/herb/vendor/bundle/ruby/3.4.0/gems/rake-13.3.0/exe/rake:27:in '<top (required)>'
/opt/hostedtoolcache/Ruby/3.4.1/x64/bin/bundle:25:in 'Kernel#load'
/opt/hostedtoolcache/Ruby/3.4.1/x64/bin/bundle:25:in '<main>'
Tasks: TOP => gem:x86_64-linux-gnu
(See full trace by running task with --trace)
Error: Process completed with exit code 1.
```


However, bumping the Ruby Version to `3.4.5` fixes the issue. 

It seems like this pull request (https://github.com/rake-compiler/rake-compiler-dock/pull/156) which is now on main was published on the old version. Though this is a bit surprising since the docker image was (re?)published using the old version number from January.

Or am I missing something?

--- a/.github/workflows/publish-images.yml
+++ b/.github/workflows/publish-images.yml
@@ -107,21 +107,10 @@ jobs:
           registry: ghcr.io
           username: ${{github.actor}}
           password: ${{secrets.GITHUB_TOKEN}}
-      - name: Generate docker image names
-        id: rcd_config
-        run: |
-          bundle exec ruby -e ' \
-            require "rake_compiler_dock"; \
-            print "snapshot_name_opt=-t "; \
-            puts RakeCompilerDock::Starter.container_image_name(:platform => %q(${{matrix.platform}}), :version => %q(snapshot)); \
-            if %q(${{matrix.platform}}).end_with?("-gnu"); \
-              print "generic_linux_snapshot_name_opt=-t "; \
-              puts RakeCompilerDock::Starter.container_image_name(:platform => %q(${{matrix.platform}}), :version => %q(snapshot)).chomp("-gnu"); \
-            end \
-          ' | tee -a $GITHUB_OUTPUT
       - name: Use cache and push docker image
         env:
-          RCD_DOCKER_BUILD: docker buildx build --cache-from=type=local,src=tmp/build-cache-X64 --cache-from=type=local,src=tmp/build-cache-ARM64 --cache-to=type=local,dest=tmp/build-cache-new ${{steps.rcd_config.outputs.snapshot_name_opt}} ${{steps.rcd_config.outputs.generic_linux_snapshot_name_opt}}
+          RCD_IMAGE_VERSION: snapshot
+          RCD_DOCKER_BUILD: docker buildx build --cache-from=type=local,src=tmp/build-cache-X64 --cache-from=type=local,src=tmp/build-cache-ARM64 --cache-to=type=local,dest=tmp/build-cache-new
         run: |
           docker buildx create --driver docker-container --use
           bundle exec rake release:${{matrix.platform}}

--- a/.github/workflows/release-images.yml
+++ b/.github/workflows/release-images.yml
@@ -110,51 +110,14 @@ jobs:
         with:
           ruby-version: "3.3"
           bundler-cache: true
-      - name: Generate docker image names
-        id: rcd_config
-        run: |
-          bundle exec ruby -e ' \
-            require "rake_compiler_dock"; \
-            print "image_name_opt=-t "; \
-            puts RakeCompilerDock::Starter.container_image_name(:platform => %q(${{matrix.platform}})); \
-            print "snapshot_name_opt=-t "; \
-            puts RakeCompilerDock::Starter.container_image_name(:platform => %q(${{matrix.platform}}), :version => %q(snapshot)); \
-            if %q(${{matrix.platform}}).end_with?("-gnu"); \
-              print "generic_linux_snapshot_name_opt=-t "; \
-              puts RakeCompilerDock::Starter.container_image_name(:platform => %q(${{matrix.platform}}), :version => %q(snapshot)).chomp("-gnu"); \
-            end \
-          ' | tee -a $GITHUB_OUTPUT
-
-
-          bundle exec ruby -e ' \
-            require "rake_compiler_dock"; \
-            print "image_name="; \
-            puts RakeCompilerDock::Starter.container_image_name(:platform => %q(${{matrix.platform}})); \
-            if %q(${{matrix.platform}}).end_with?("-gnu"); \
-              print "generic_linux_image_name="; \
-              puts RakeCompilerDock::Starter.container_image_name(:platform => %q(${{matrix.platform}})).chomp("-gnu"); \
-            end \
-          ' | tee -a $GITHUB_OUTPUT
-      - name: Build docker image
-        env:
-          RCD_DOCKER_BUILD: docker buildx build --cache-from=type=local,src=tmp/build-cache --cache-to=type=local,dest=tmp/build-cache-new
-        run: |
-          docker buildx create --driver docker-container --use
-          bundle exec rake build:x86:${{matrix.platform}}
-          # move build cache and remove outdated layers
-          rm -rf tmp/build-cache
-          mv tmp/build-cache-new tmp/build-cache
       - uses: docker/login-action@v3
         with:
           registry: ghcr.io
           username: ${{github.actor}}
           password: ${{secrets.GITHUB_TOKEN}}
-      - name: Push the docker image
+      - name: Use cache and push docker image
+        env:
+          RCD_DOCKER_BUILD: docker buildx build --cache-from=type=local,src=tmp/build-cache-X64 --cache-from=type=local,src=tmp/build-cache-ARM64 --cache-to=type=local,dest=tmp/build-cache-new
         run: |
-          docker images
-          docker push ${{steps.rcd_config.outputs.image_name}}
-      - name: Push a generic linux image
-        if: ${{ steps.rcd_config.outputs.generic_linux_image_name }}
-        run: |
-          docker tag ${{steps.rcd_config.outputs.image_name}} ${{steps.rcd_config.outputs.generic_linux_image_name}}
-          docker push ${{steps.rcd_config.outputs.generic_linux_image_name}}
+          docker buildx create --driver docker-container --use
+          bundle exec rake release:${{matrix.platform}}

--- a/README.md
+++ b/README.md
@@ -315,6 +315,7 @@ OCI images snapshotted from `main` are published weekly to Github Container Regi
 - `ghcr.io/rake-compiler/rake-compiler-dock-image:snapshot-mri-x86_64-linux-gnu`
 
 These images are intended for integration testing. They may not work properly and should not be considered production ready.
+Snapshot images can be used by setting `RCD_IMAGE_VERSION` environment variable.
 
 
 ## Environment Variables
@@ -330,6 +331,9 @@ The following variables are recognized by rake-compiler-dock:
     It is ignored when `rubyvm` is set to `:jruby`.
 * `RCD_IMAGE` - The docker image that is downloaded and started.
     Defaults to "ghcr.io/rake-compiler/rake-compiler-dock-image:IMAGE_VERSION-PLATFORM" with an image version that is determined by the gem version.
+* `RCD_IMAGE_VERSION` - The docker image version that is downloaded and started.
+    Defaults to the version of the rake-compiler-dock gem in use.
+    Can be set to `snapshot` in order to use the latest images from main branch.
 
 The following variables are passed through to the docker container without modification:
 

--- a/lib/rake_compiler_dock/starter.rb
+++ b/lib/rake_compiler_dock/starter.rb
@@ -167,7 +167,7 @@ module RakeCompilerDock
 
           "%s/rake-compiler-dock-image:%s-%s%s" % [
             container_registry,
-            options.fetch(:version) { IMAGE_VERSION },
+            options.fetch(:version) { image_version },
             container_rubyvm(options),
             container_jrubyvm?(options) ? "" : "-#{options.fetch(:platform)}",
           ]
@@ -176,6 +176,10 @@ module RakeCompilerDock
 
       def container_registry
         ENV['CONTAINER_REGISTRY'] || "ghcr.io/rake-compiler"
+      end
+
+      def image_version
+        ENV['RCD_IMAGE_VERSION'] || IMAGE_VERSION
       end
 
       def container_rubyvm(options={})

--- a/test/test_starter.rb
+++ b/test/test_starter.rb
@@ -38,6 +38,12 @@ class TestStarter < Test::Unit::TestCase
     with_env({"RAKE_COMPILER_DOCK_IMAGE" => "env-var-value"}) do
       assert_equal("env-var-value", Starter.container_image_name)
     end
+    with_env({"CONTAINER_REGISTRY" => "env-var-value"}) do
+      assert_match(/env-var-value\//, Starter.container_image_name(platform: "x64-mingw-ucrt"))
+    end
+    with_env({"RCD_IMAGE_VERSION" => "env-var-value"}) do
+      assert_match(/:env-var-value-/, Starter.container_image_name(platform: "x64-mingw-ucrt"))
+    end
 
     # with image option
     assert_equal("option-value", Starter.container_image_name({:image => "option-value"}))

